async-nats/src/jetstream/types.rs

```diff
@@ -0,0 +1,193 @@
+use time::serde::rfc3339;
+
+/// `StreamConfig` determines the properties for a stream.
+/// There are sensible defaults for most. If no subjects are
+/// given the name will be used as the only subject.
+#[derive(Debug, Default, Serialize, Deserialize, Clone, PartialEq, Eq)]
+pub struct StreamConfig {
+    /// A name for the Stream. Must not have spaces, tabs or period `.` characters
+    pub name: String,
+    /// How large the Stream may become in total bytes before the configured discard policy kicks in
+    pub max_bytes: i64,
+    /// How large the Stream may become in total messages before the configured discard policy kicks in
+    pub max_msgs: i64,
+    /// Maximum amount of messages to keep per subject
+    pub max_msgs_per_subject: i64,
+    /// When a Stream has reached its configured `max_bytes` or `max_msgs`, this policy kicks in.
+    /// `DiscardPolicy::New` refuses new messages or `DiscardPolicy::Old` (default) deletes old messages to make space
+    pub discard: DiscardPolicy,
+    /// Which NATS subjects to populate this stream with. Supports wildcards. Defaults to just the
+    /// configured stream `name`.
+    #[serde(default, skip_serializing_if = "Vec::is_empty")]
+    pub subjects: Vec<String>,
+    /// How message retention is considered, `Limits` (default), `Interest` or `WorkQueue`
+    pub retention: RetentionPolicy,
+    /// How many Consumers can be defined for a given Stream, -1 for unlimited
+    pub max_consumers: i32,
+    /// Maximum age of any message in the stream, expressed in nanoseconds
+    #[serde(with = "serde_nanos")]
+    pub max_age: Duration,
+    /// The largest message that will be accepted by the Stream
+    #[serde(default, skip_serializing_if = "is_default")]
+    pub max_msg_size: i32,
+    /// The type of storage backend, `File` (default) and `Memory`
+    pub storage: StorageType,
+    /// How many replicas to keep for each message in a clustered JetStream, maximum 5
+    pub num_replicas: usize,
+    /// Disables acknowledging messages that are received by the Stream
+    #[serde(default, skip_serializing_if = "is_default")]
+    pub no_ack: bool,
+    /// The window within which to track duplicate messages.
+    #[serde(default, skip_serializing_if = "is_default")]
+    pub duplicate_window: i64,
+    /// The owner of the template associated with this stream.
+    #[serde(default, skip_serializing_if = "is_default")]
+    pub template_owner: String,
+    /// Indicates the stream is sealed and cannot be modified in any way
+    #[serde(default, skip_serializing_if = "is_default")]
+    pub sealed: bool,
+    #[serde(default, skip_serializing_if = "is_default")]
+    /// A short description of the purpose of this stream.
+    pub description: Option<String>,
+    #[serde(
+        default,
+        rename = "allow_rollup_hdrs",
+        skip_serializing_if = "is_default"
+    )]
+    /// Indicates if rollups will be allowed or not.
+    pub allow_rollup: bool,
```

Comment:
Do we wanna be explicit?

```suggestion
    pub allow_rollup_headers: bool,
```

Reply:
No. I would stick to name describing behaviour.